handlers/get_questions.go

```diff
@@ -0,0 +1,122 @@
+// @Router /getquestions [get]
+func (h *Handler) GetQuestions(c echo.Context) error {
+	// Initialize request and response objects
+	getQuestionsRequest := new(bindings.BenefitQuestionsRequest)
+	getQuestionsResponse := new(renderings.BenefitQuestionsResponse)
+
+	// Bind incoming data to request object
+	if err := c.Bind(getQuestionsRequest); err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, err)
+	}
+
+	// Copy request field values to response
+	getQuestionsResponse.RequestDate = getQuestionsRequest.RequestDate
+	getQuestionsResponse.LifeJourneys = getQuestionsRequest.LifeJourneys
+	getQuestionsResponse.BenefitList = getQuestionsRequest.BenefitList
+
+	// Convert UNIX timestamp to required OpenFisca time value format
+	t := time.Unix(getQuestionsRequest.RequestDate, 0)
+	tstring := t.Format("2006-01-02")
+
+	// Initialize top level "Persons" map
+	persons := map[string]interface{}{}
+
+	// Initialize second level "Person" map
+	p := map[string]map[string]interface{}{}
+
+	// Get benefits from client request and format for request to OpenFisca
+	for _, benefit := range getQuestionsRequest.BenefitList {
+		p[benefit] = map[string]interface{}{}
+		p[benefit][tstring] = nil
+	}
+	persons["persons"] = Persons{p}
+
+	// Send request to OpenFisca trace endpoint to find dependencies of requested benefits
+	ofResponse, err := openfisca.OFService.SendRequest(openfisca.OFService{}, persons)
+	if err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, err)
+	}
+
+	// Get trace object out of the OpenFisca response
+	trace := ofResponse["trace"].(map[string]interface{})
+
+	// Load questions from local file
+	questionList, err := questions.Service.LoadQuestions()
+	if err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, err)
+	}
+	
+	// Loop through OpenFisca trace response
+	for benefitKey, benefitObject := range trace {
+		// Find benefit keys
+		if strings.Contains(benefitKey, "is_eligible") {
+			// Remove date from benefit key string
+			benefitKey = benefitKey[:strings.IndexByte(benefitKey, '<')]
+			// Loop through benefit's child object
+			for benefitInfoKey, benefitInfo := range benefitObject.(map[string]interface{}) {
+				// Find array of benefit dependencies
+				if strings.Contains(benefitInfoKey, "dependencies") {
+					// For each dependency in array
+					for _, dependency := range benefitInfo.([]interface{}) {
+						// Loop through Question List
+						for key, question := range questionList {
+							// Find question that references the current dependency of the current benefit
+							if contains(question.OpenFiscaIds, dependency.(string)[:strings.IndexByte(dependency.(string), '<')]) {
+								// Append the current benefit to the DependencyOf array for the current Question
+								question.DependencyOf = append(question.DependencyOf, benefitKey)
+								questionList[key] = question
+							}
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
This is hard to read and some of the functionality could be common for the OFService. ie... get is_eligible values